Chaincase.iOS/Tor/TorProcessManager.cs

```diff
@@ -48,6 +49,8 @@ public OnionManager()
 
         public static bool RequestFallbackAddressUsage { get; private set; } = false;
 
+        private readonly AsyncLock mutex = new AsyncLock();
```

Comment:
This mutex is specific for functions in this file. Some of them have to be async so they need their own mutex

Reply:
``// mutex prevents conflicts between async callbacks and stopping Tor